File: frontend/src/types/notification.ts
// ============================================================
// Notification Types
// ============================================================

export interface Notification {
  id: number;
  user_id: number;
  project_id: number | null;
  challenge_id?: number | null;
  notification_type: string;
  priority: "LOW" | "MEDIUM" | "HIGH" | "URGENT";
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
  read_at: string | null;
}

File: mediator_backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    project_id: int | None

    notification_type: str
    priority: str

    title: str
    message: str

    is_read: bool

    created_at: datetime
    read_at: datetime | None

class NotificationMarkRead(BaseModel):
    pass